### PR TITLE
feat: add first endpoint call, types, validation, errors, tests and docs

### PR DESCRIPTION
**Pull request by oxcabe**
## Description
This PR settles the base structure and functionality needed to implement calling logic for any Etherscan API endpoint.

The following features are added:
* Type definitions for the structure of the API (modules, actions, calls, requests, responses...).
* In-source testing configuration.
* Partial documentation of the current state of the project.
* Documentation generating tools.
* Custom errors.
* Runtime validation schemas to parse, validate and transform API call responses.
* The implementation of the `account/balance` endpoint call, which uses all the features above and can be taken as reference for implementing the remaining endpoint calls.
* Test coverage functionality.
* A CONTRIBUTING file with instructions on how to set-up the development environment.

### CODE REVIEW

File: src/account/index.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * Actions inside the `account` module.
+ */
+export type AccountModuleActions = BalanceAction;
+
+/**
+ * Request base for `account` module actions.
+ */
+export type AccountModuleRequest = EtherscanModuleRequest<"account">;
```
Comment (EN2U) [inline src/account/index.ts:12]: I think we should add "account" string as a constant

Comment (oxcabe) [reply src/account/index.ts:12]: I agree

Comment (oxcabe) [reply src/account/index.ts]: Added in [a17c665](https://github.com/railmap/etherscan/pull/3/commits/a17c66532911ff85bc63dd0474bb01a37325c0ff).

**Review by EN2U — CHANGES_REQUESTED** [review]
Great code! in general i just simply suggest to add the strings as "address" to add them as constants. This ensures that no one gets the syntax or spelling wrong.

For tests, we could also separate the checks into functions. That is to say, the it() section could be extracted in a function so that the tests do not scale and are easier to read.

File: src/account/balance/call.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+if (import.meta.vitest !== undefined) {
+  const { it, expect, describe } = import.meta.vitest;
+  const { InvalidAddressFormatError } = await import("etherscan/types");
+
+  describe("balance", () => {
+    const balanceParams = {
+      address: process.env.SEPOLIA_ADDRESS ?? "",
+      tag: "latest",
+      apiKey: process.env.ETHERSCAN_API_KEY ?? "",
+    };
```
[the remark this answers is not in the document]
Comment (oxcabe) [reply src/account/balance/call.ts:55]: I agree

File: src/account/balance/call.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      const invalidAddressParams = { ...balanceParams, address: "invalid" };
+      const { error } = await balance(
+        EtherscanBaseUrl.Sepolia,
+        invalidAddressParams,
+      );
+
+      expect(error).toBeInstanceOf(InvalidAddressFormatError);
+    });
+  });
+}
```
[the remark this answers is not in the document]
Comment (oxcabe) [reply src/account/balance/call.ts:79]: Yes, it's just not necessary for the current state of the project. It is fair to have this in mind for the future, though.

File: src/errors.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      it("should return a Error derived class", () => {
+        expect(testError).toBeInstanceOf(Error);
+      });
+
+      it("should return a class that is convertible to string", () => {
+        expect(String(testError)).toStrictEqual("Error: Test error");
+      });
+    });
+  });
+}
```
Comment (EN2U) [inline src/errors.ts:59]: we should discuss this about using a [decorator](https://stackoverflow.com/a/46187742)

**Review by EN2U — APPROVED** [review]
Everything is OK!

We have to talk about the decorator usage for handling errors but we can solve it in another pr

**Pull request merged**